thirdparty/runtime_environments.go

```diff
@@ -258,28 +259,35 @@ func (c *RuntimeEnvironmentsClient) GetToolchains(ctx context.Context, opts Tool
 	return toolchains, nil
 }
 
-// ImageDiffOptions represents the arguments for getImageDiff. AMIBefore is the starting AMI, and AMIAfter is the ending AMI.
-type ImageDiffOptions struct {
-	AMIBefore string `json:"-"`
-	AMIAfter  string `json:"-"`
+// APIDiffResponse represents a response from the /rest/api/v1/ami/diff route.
+type APIDiffResponse struct {
+	Data          []ImageDiffChange `json:"data"`
+	FilteredCount int               `json:"filtered_count"`
+	TotalCount    int               `json:"total_count"`
 }
 
 // ImageDiffChange represents a change between two AMIs.
 type ImageDiffChange struct {
-	Name    string `json:"name"`
-	Manager string `json:"manager"`
-	Type    string `json:"type"`
-	Removed string `json:"removed"`
-	Added   string `json:"added"`
+	AfterVersion  string `json:"after_version"`
+	BeforeVersion string `json:"before_version"`
+	Name          string `json:"name"`
+	Manager       string `json:"manager"`
+	Type          string `json:"type"`
+}
+
+// ImageDiffOptions represents the arguments for getImageDiff. AMIBefore is the starting AMI, and AMIAfter is the ending AMI.
+type ImageDiffOptions struct {
+	AMIBefore string `json:"-"`
```

Comment:
If we aren't using this struct for json, then we can just omit the tags entirely. We may also want to make this a private struct.

Reply:
I think I'll keep the tags because Kim gave me some advice to add it before [here](https://github.com/evergreen-ci/evergreen/pull/8193#discussion_r1717488780)! I think it helps to let whoever is reading the code to know that it isn't meant to be used in a JSON payload